do not pass dsc_timeout as timeout parameter to DSC resource params

schlitzered opened the pull request:
## Summary
dsc_timeout is used to set the timeout on how long puppet/powershell will wait on the execution of a command.
dsc_timeout is also passed as "timeout" to the DSC resource parameters itself.

but it seems like DSC resources do not have a timout parameter.

see also: https://github.com/puppetlabs/ruby-pwsh/issues/358


## Related Issues (if any)
https://github.com/puppetlabs/ruby-pwsh/issues/358


## Checklist
- [ ] 🟢 Spec tests.
- [ ] 🟢 Acceptance tests.
- [x] Manually verified.

sry, i am not a ruby guy, and i have no clue on how to write Spec/Acceptance tests.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/puppetlabs/ruby-pwsh?pullRequest=359) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/puppetlabs/ruby-pwsh?pullRequest=359) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/puppetlabs/ruby-pwsh?pullRequest=359) it.</sub>

LukasAud commented:
Hi @schlitzered, it looks like there is a few rubocop errors in this PR, could you address them?

schlitzered commented:
> Hi @schlitzered, it looks like there is a few rubocop errors in this PR, could you address them?

well the only check that complained was 43 instead of 42 lines in one function. i just removed one of the two comments i have added, and now the functions is back to 42 lines, and the check is passing.

schlitzered commented:
okay, it is still failing. no clue why, the actual logic is only one line, if have no clue how i should make this any shorter.

jordanbreen28 commented:
@schlitzered in niche cases like this it's absolutely fine to just add an exclusion, try running
`rubocop --auto-gen-config` and push up the modified rubocop_todo

jordanbreen28 commented:
@schlitzered thanks for picking up on this and submitting the PR to get the ball rolling, I've raised https://github.com/puppetlabs/ruby-pwsh/pull/366 rather than to go back and forth with you for rubocop fixes 🙂
